Doc/c-api/code.rst

```diff
@@ -30,10 +30,14 @@ bound into a function.
    Return true if *co* is a :ref:`code object <code-objects>`.
    This function always succeeds.
 
-.. c:function:: int PyCode_GetNumFree(PyCodeObject *co)
+.. c:function:: Py_ssize_t PyCode_GetNumFree(PyCodeObject *co)
 
    Return the number of free variables in *co*.
 
+.. c:function:: int PyCode_GetFirstFree(PyCodeObject *co)
+
+   Return the number of local + cell variables in *co*.
```

Comment:
Document what the API does, not its implementation details: `PyCode_GetFirstFree` returns the position (or offset) of the first free variable in *co*.

FTR: @markshannon and @iritkatriel introduced this API in gh-100721. It is available via Python.h.

Reply:
Done. But my question above still stands. If I'm right I might make a separate pull request for this.

Reply:
See the PR I linked to, especially https://github.com/python/cpython/pull/100721#discussion_r1061391884 and 49ca0446ac0becf20a748a0800c788727993aec0.

Reply:
Thank you. If there is anything else here that I could fix, I'll be happy to do it.

Reply:
I would still like to hear from either @markshannon or @iritkatriel; I'm not sure this was intended as a public API.

Reply:
It was and it wasn't. We don't really want to expose it, but if we don't, then Cython and other C extensions will access the internal fields directly, which is worse.

Now that we have an unstable API, it should probably be part of that.
I'd like all code object C APIs to be unstable, but it's probably too late for that.

Maybe rename it to `PyUnstableCode_GetFirstFree`?

Reply:
> Maybe rename it to PyUnstableCode_GetFirstFree?

Sounds like a good idea, but since it is already included in 3.12, we have to deprecate `PyCode_GetFirstFree` and `PyUnstableCode_GetFirstFree` in 3.13.

Reply:
Follow-up issue: #115756.